transaction_receipt_query.go

```diff
@@ -271,17 +271,12 @@ func (query *TransactionReceiptQuery) Execute(client *Client) (TransactionReceip
 
 	if err != nil {
 		if precheckErr, ok := err.(ErrHederaPreCheckStatus); ok {
-			receipt := TransactionReceipt{}
-			if resp.query.GetTransactionGetReceipt() != nil {
-				receipt = _TransactionReceiptFromProtobuf(resp.query.GetTransactionGetReceipt().GetReceipt())
-			}
-
-			return receipt, ErrHederaReceiptStatus{
-				TxID:    precheckErr.TxID,
-				Status:  precheckErr.Status,
-				Receipt: receipt,
-			}
+			return TransactionReceipt{}, _NewErrHederaReceiptStatus(precheckErr.TxID, precheckErr.Status)
 		}
+		if respErr, ok := err.(ErrHederaReceiptStatus); ok {
+			return respErr.Receipt, nil
```

Comment:
In the result of an error we should return both, a filled out receipt and an error.

Reply:
In my opinion, getting a transaction's receipt should not resolve with an error, just because the transaction did not result with SUCCESS. An error would be something that happens when you fail to get the receipt (e.g. timeout rpc call). I think that two things are mixed here  - having a non SUCCESSFUL transaction and an actual error while trying to get the receipt.

Reply:
From a developer standpoint, if this return an error, I have to always check that the error is `exceptional receipt status: IDENTICAL_SCHEDULE_ALREADY_CREATED`, so that I am sure that the receipt is filled.

Reply:
Yea, that sounds fair.